src/index/gpu_raft/gpu_raft_brute_force.cc

```diff
@@ -21,8 +21,17 @@
 #include "gpu_raft.h"
 #include "knowhere/factory.h"
 #include "knowhere/index_node_thread_pool_wrapper.h"
+#include "raft/util/cuda_rt_essentials.hpp"
 
 namespace knowhere {
-KNOWHERE_REGISTER_GLOBAL_WITH_THREAD_POOL(GPU_RAFT_BRUTE_FORCE, GpuRaftBruteForceIndexNode, fp32, cuda_concurrent_size);
-KNOWHERE_REGISTER_GLOBAL_WITH_THREAD_POOL(GPU_BRUTE_FORCE, GpuRaftBruteForceIndexNode, fp32, cuda_concurrent_size);
+KNOWHERE_REGISTER_GLOBAL_WITH_THREAD_POOL(GPU_RAFT_BRUTE_FORCE, GpuRaftBruteForceIndexNode, fp32, []() {
+    int count;
+    RAFT_CUDA_TRY(cudaGetDeviceCount(&count));
+    return count * 8;
```

Comment:
Please put this magic number as a constant in some header file, coupled with comments about why it is `8` at this moment :)